precompiles/erc20/approve.go

```diff
@@ -264,8 +322,11 @@ func (p Precompile) decreaseAllowance(
 	}
 
 	amount = new(big.Int).Sub(allowance.Amount.BigInt(), subtractedValue)
+	if amount.Sign() < 0 {
+		return nil, fmt.Errorf("subtracted value cannot be greater than existing allowance: %s > %s", subtractedValue, allowance.Amount)
```

Comment:
let's check for this error too

Reply:
This is only a safety check here, since this function is only called if `subtractedValue < allowance`